packages/discussions/src/utils/request.ts

```diff
@@ -0,0 +1,73 @@
+import { IRequestOptions } from "../types";
+
+/**
+ * returns Promise that resolves token to use in Discussions API requests
+ *
+ * @export
+ * @param {IRequestOptions} options
+ * @return {*}  {Promise<string>}
+ */
+export function authenticateRequest(options: IRequestOptions): Promise<string> {
+  const { token, authentication } = options;
+
+  let tokenPromise = () => {
+    return Promise.resolve(token);
+  };
+
+  if (authentication) {
+    tokenPromise = authentication.getToken.bind(
+      authentication,
+      authentication.portal
+    );
+  }
+
+  return tokenPromise();
+}
+
+/**
+ * parses IRequestOptions and makes request against Discussions API
+ *
+ * @export
+ * @param {string} url
+ * @param {IRequestOptions} options
+ * @param {string} [token]
+ * @return {*}  {Promise<any>}
+ */
+export function apiRequest(
+  url: string,
+  options: IRequestOptions,
+  token?: string
+): Promise<any> {
+  const headers = new Headers();
+  headers.append("Content-Type", "application/json");
+  if (token) {
+    headers.append("Authorization", `Bearer ${token}`);
+  }
+
+  const opts: RequestInit = {
+    headers,
+    method: options.method || "GET",
+    mode: options.mode || "cors",
+    cache: options.cache || "no-cache",
+    credentials: options.credentials || "include"
+  };
+
+  const apiBase = options.apiBaseUrl || "http://localhost/api/v1";
```

Comment:
do we really want to default to localhost? is `/api/v1` even valid?

Reply:
suppose this could be a required param with no default

Reply:
I think in other places we've defaulted to the prod URL, but I've worried about that leading to subtle bugs.

Reply:
we don't have a prod url at the moment is the only issue here, so until we do we can leave it a required param